### PR TITLE
Simplify DAPJS use

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
If we switch to WebSerial we'll just do that, not have a common abstraction.

Rob and I have confirmed this works in practice.

With both this and main, with some degree of performance throttling, you can see an occasional deviation from a flat line if you just watch and wait but that's not new in this PR (maybe bug, maybe fundamental perf. issue, TBD).

### CODE REVIEW

File: src/script/microbit-interfacing/MicrobitSerial.ts [hunk 1]
```diff
@@ -47,99 +41,27 @@ class MicrobitSerial implements MicrobitConnection {
...
+        if (this.serialProtocolState === SerialProtocolState.AwaitingHandshakeResponse) {
+          let handshakeResponse = protocol.processHandshake(msg);
+          if (handshakeResponse && handshakeResponse.value === protocol.version) {
+            this.serialProtocolState = SerialProtocolState.Running;
+
+            // Request the micro:bit to start sending the periodic messages
+            const startCmd = protocol.generateCommand(protocol.CommandTypes.Start);
+            await this.usb.serialWrite(startCmd.message);
+            // TODO: Unclear why we need to send this command twice, need to investigate
+            // await this.usb.serialWrite(startCmd.message);
```
Comment (microbit-matt-hillsdon) [inline src/script/microbit-interfacing/MicrobitSerial.ts:61]:
```suggestion
```

This doesn't appear to be necessary.

File: src/script/microbit-interfacing/MicrobitSerial.ts [hunk 2]
```diff
@@ -47,99 +41,27 @@ class MicrobitSerial implements MicrobitConnection {
...
+          }
+        } else {
+          // Running :)
```
Comment (microbit-matt-hillsdon) [inline src/script/microbit-interfacing/MicrobitSerial.ts:64]:
```suggestion
```

File: src/script/microbit-interfacing/MicrobitSerial.ts [hunk 3]
```diff
@@ -161,49 +80,47 @@ class MicrobitSerial implements MicrobitConnection {
...
-  }
-
   public isConnected(): boolean {
-    return this.connected;
+    return this.usb.isSerialConnected();
   }
 
   public disconnect(): void {
-    this.connected = false;
+    // Weirdly this disconnects the CortexM...
```
Comment (microbit-matt-hillsdon) [inline src/script/microbit-interfacing/MicrobitSerial.ts:116]: I want to come back to this. I don't think MicrobitUSB should extend CortexM and I doubt this disconnect is intentional/necessary.

File: src/script/microbit-interfacing/MicrobitUSB.ts [hunk 4]
```diff
@@ -122,8 +125,35 @@ class MicrobitUSB extends CortexM {
...
+    await this.transport.open();
+
+    this.serialDAPLink = new DAPLink(this.transport);
+    const initialBaudRate = await this.serialDAPLink.getSerialBaudrate();
+    this.serialDAPLink.addListener(DAPLink.EVENT_SERIAL_DATA, callback);
+    await this.serialDAPLink.connect();
+    if (initialBaudRate !== baudRate) {
+      await this.serialDAPLink.setSerialBaudrate(baudRate);
+    }
+    this.serialPromise = this.serialDAPLink.startSerialRead(5, false);
```
Comment (microbit-carlos) [inline src/script/microbit-interfacing/MicrobitUSB.ts:138]: Is this configuring the pulling rate to 5ms? Maybe we should set that as a local const like baudRate?

File: src/script/microbit-interfacing/MicrobitSerial.ts (hunk 3 above)
Comment (microbit-carlos) [reply src/script/microbit-interfacing/MicrobitSerial.ts:116]: Is that a question about extending something a bit more high level like the `DAPLink` class, or that it shouldn't extend anything and use composition instead?

File: src/script/microbit-interfacing/MicrobitUSB.ts (hunk 4 above)
Comment (microbit-matt-hillsdon) [reply src/script/microbit-interfacing/MicrobitUSB.ts:138]: Yeah sure I'll pull it out. 

Value TBD as we test, but I figure it should be more > 2x more frequent than the writes to get timely data.

We'll follow up with some code to log periodic bucketed performance stats as the console.logs seem to have something of an effect on performance.

File: src/script/microbit-interfacing/MicrobitSerial.ts (hunk 3 above)
Comment (microbit-matt-hillsdon) [reply src/script/microbit-interfacing/MicrobitSerial.ts:116]: Not convinced it should extend anything. But will review on a separate PR.

File: src/script/microbit-interfacing/MicrobitUSB.ts (hunk 4 above)
Comment (microbit-matt-hillsdon) [reply src/script/microbit-interfacing/MicrobitUSB.ts]: 38a1cb4e4e847bbd8f5b826f619b739c7083a518

**Comment by microbit-matt-hillsdon** [comment]
@microbit-carlos I'm going to go ahead and merge this to help us make progress. Let me know if you have any more feedback.

**Pull request merged**